Bug #14410 [Collecte] - UA types "Static Attachment" and "Dynamic Attachment" should no longer appear in the Collection panels.

Salimdev opened the pull request:
Les UA de type "Static Attachment" et "Dynamic Attachment" continuent d'apparaître dans les panneaux de Collecte

Commit:
Bug #14410 [Collecte] - UA types "Static Attachment" and "Dynamic Attachment" should no longer appear in the Collection panels.

vitam-prg commented:
![Logo](https://cdn.ast.checkmarx.net/integrations/logo/Checkmarx.png)
**Checkmarx One – Scan Summary & Details** – [c35047f5\-f83b\-408b\-bcf8\-42fbbb379a54](https://eu.ast.checkmarx.net/projects/3f06ff53-d733-40ac-a6fb-49e415cb793f/scans?id=c35047f5-f83b-408b-bcf8-42fbbb379a54&branch=bug_14410)
### Great job, no security vulnerabilities found in this Pull Request

bbenaissa commented on api/api-collect/collect/src/main/java/fr/gouv/vitamui/collect/server/service/TransactionArchiveUnitService.java:
```diff
@@ -178,6 +183,17 @@ public VitamUIArchiveUnitResponseDto searchArchiveUnitsByCriteria(
+            .noneMatch(ctr -> ctr.getCriteria().startsWith(UPDATE_OPERATION_FIELD));
+        if (searchWithEmptyUpdateOperationField) {
+            /* Ensure that we do not return unit having an UpdateOperation value in the results */
+            Query originalQuery = searchQuerySelectMultiQuery.getQueries().get(0);
```
Je ne sais pas si c'est possible d'avoir une query vide ? je pense que c'est possible, peut etre que je me trompe, si c'est le cas, à changer la condition

marob committed:
Fix #14430: tree-node doesn't accept null date

- set data to [] instead of null
- code refactoring + remove unused code
- refreshTreeNodes after all responses arrived instead of after each one

Commit:
Bug #14410 [Collecte] - UA types "Static Attachment" and "Dynamic Attachment" should no longer appear in the Collection panels.
